src/mgmt/srv_internal.h

```diff
@@ -95,6 +96,43 @@ int ds_mgmt_destroy_pool(uuid_t pool_uuid, const char *group, uint32_t force);
 void ds_mgmt_hdlr_pool_create(crt_rpc_t *rpc_req);
 void ds_mgmt_hdlr_pool_destroy(crt_rpc_t *rpc_req);
 
+/** srv_query.c */
+
+/* Device health stats from bio_dev_state */
+struct mgmt_bio_health {
+	struct smd_dev_info	*mb_dev_info;
+	uuid_t			 mb_devid;
+	uint64_t		*mb_media_errors;
+	uint64_t		 mb_error_count;
+	uint32_t		 mb_read_errs;
+	uint32_t		 mb_write_errs;
+	uint32_t		 mb_unmap_errs;
+	uint16_t		 mb_temperature;
+	bool			 mb_temp;
+	bool			 mb_spare;
+	bool			 mb_readonly;
+	bool			 mb_reliability;
+	bool			 mb_volatile;
+};
+
+int ds_mgmt_bio_health_query(struct mgmt_bio_health *mbh, uuid_t uuid,
+			     char *tgt_id);
+
+struct mgmt_smd_device {
+	uuid_t			 devid;
+	int			*tgt_ids;
+	uint32_t		 tgt_cnt;
+	struct mgmt_smd_device	*next;
+};
+
+struct mgmt_smd_devs {
+	struct mgmt_smd_device *ms_devs;
+	struct mgmt_smd_device *ms_head;
+	int			ms_num_devs;
+};
```

Comment:
Dup of smd_dev_info?

Reply:
I can again simplify this to include a smd_dev_info pointer, but I still need to track the pointers to head and next, etc, as well as the number of devs in the list. This could possibly be done just with smd_dev_info, but might be more complex with d_list?

Reply:
I see, if it's superset of smd_dev_info, you can include smd_dev_info in this structure, and I think we'd better to use d_list instead of implementing another specific list. (Not quite sure the complexity you mentioned about using d_list).

Reply:
ok will just use smd_dev_info, just had to revise the code a bit and passed resp instead of a mgmt struct. I thought I would have to still maintain another struct besides the Mgmt__SmdDevResp (that's why I thought there would need to be more code with d_list), but doesn't seem to be